presto-hive/src/main/java/com/facebook/presto/hive/HiveSplitManager.java

```diff
@@ -281,6 +291,49 @@ public ConnectorSplitSource getSplits(
                 layout.getPredicateColumns(),
                 layout.getDomainPredicate().getDomains());
 
+        double ratio = 1.0;
+
+        if (isDynamicSplitSizesEnabled(session)) {
+            HiveTableHandle hiveTableHandle = new HiveTableHandle(tableName.getSchemaName(), tableName.getTableName());
+            List<HiveColumnHandle> allColumnHandles = new ArrayList<>();
+            allColumnHandles.addAll(getRegularColumnHandles(table));
+            allColumnHandles.addAll(getPartitionKeyColumnHandles(table));
+            Map<String, Type> allColumnTypes = allColumnHandles.stream().collect(toImmutableMap(HiveColumnHandle::getName, handle -> metadata.getColumnMetadata(session, hiveTableHandle, handle).getType()));
+            Map<String, ColumnHandle> allColumns = allColumnHandles.stream().collect(toImmutableMap(HiveColumnHandle::getName, identity()));
+
+            Set<HiveColumnHandle> readColumnHandles = new HashSet<>();
+            if (layout.getRequestedColumns().isPresent()) {
+                readColumnHandles.addAll(layout.getRequestedColumns().get());
+            }
+            readColumnHandles.addAll(layout.getPredicateColumns().values());
+            readColumnHandles.removeIf(handle -> handle.isHidden());
+
+            TableStatistics tableStatistics = metadata.getHiveStatisticsProvider().getTableStatistics(session, tableName, allColumns, allColumnTypes, partitions);
+            double totalSize = tableStatistics.getTotalSize().getValue();
+            double requiredSize = 0;
+            double rowCount = tableStatistics.getRowCount().getValue();
+            for (Map.Entry<ColumnHandle, ColumnStatistics> entry : tableStatistics.getColumnStatistics().entrySet()) {
```

Comment:
Here we loop over all columns (can be thousands), but process only a set of readColumnHandles columns (can be a handful). Would it make sense to change this to loop over readColumnHandles instead?

Reply:
Refactored to only query for read columns. I previously incorrectly assumed totalSize for summed for just columns provided